internal/webhooks/test/clusterclass_test.go

```diff
@@ -79,30 +79,35 @@ func TestClusterClassWebhook_Fail_Create(t *testing.T) {
 	ns, err := env.CreateNamespace(ctx, "test-topology-clusterclass-webhook")
 	g.Expect(err).ToNot(HaveOccurred())
 
+	cleanup, err := createTemplates(ns)
+	g.Expect(err).ToNot(HaveOccurred())
+
 	// Create the objects needed for the integration test:
 	// - a ClusterClass with all the related templates
 	clusterClass1 := builder.ClusterClass(ns.Name, "class1").
 		WithInfrastructureClusterTemplate(
-			builder.InfrastructureClusterTemplate(ns.Name, "inf").Build()).
+			builder.TestInfrastructureClusterTemplate(ns.Name, "inf").Build()).
 		// ControlPlaneTemplate not defined in this ClusterClass making it invalid.
 		WithWorkerMachineDeploymentClasses(
 			*builder.MachineDeploymentClass("md1").
 				WithInfrastructureTemplate(
-					builder.InfrastructureMachineTemplate(ns.Name, "OLD_INFRA").Build()).
+					builder.TestInfrastructureMachineTemplate(ns.Name, "OLD_INFRA").Build()).
 				WithBootstrapTemplate(
-					builder.BootstrapTemplate(ns.Name, "bootstrap1").Build()).
+					builder.TestBootstrapTemplate(ns.Name, bootstrapTemplateName1).Build()).
 				Build(),
 			*builder.MachineDeploymentClass("md2").
 				WithInfrastructureTemplate(
-					builder.InfrastructureMachineTemplate(ns.Name, "infra1").Build()).
+					builder.TestInfrastructureMachineTemplate(ns.Name, infrastructureClusterTemplateName1).Build()).
```

Comment:
infrastructureClusterTemplateName1 for TestInfrastructureMachineTemplate seems off